src/main/java/org/checkerframework/checker/noliteral/NoLiteralAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,42 @@
+package org.checkerframework.checker.noliteral;
+
+import javax.lang.model.element.AnnotationMirror;
+import org.checkerframework.checker.noliteral.qual.MaybeConstant;
+import org.checkerframework.checker.noliteral.qual.NonConstant;
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.javacutil.AnnotationBuilder;
+
+/**
+ * The type factory for the no literal checker. It doesn't do much: all it provides are some
+ * canonical annotations. If we want to change this checker so that it uses the Value Checker for
```

Comment:
Say why.  For example, would that reduce false negatives (improve soundness) or false positives (improve precision)?
I think the point is to identify more possible constant values than just literals in the source code.